addons/nodot/interaction/Teleport3D.gd

```diff
@@ -0,0 +1,88 @@
+## Teleport3D node
+class_name Teleport3D extends Area3D
+
+@export_category("Nodes")
+## Target node for copy the coords
+@export_node_path("Node3D") var Target
+## Player node or other objective to teleport
+@export_node_path("CharacterBody3D") var Player
+
+@export_category("Others Variables")
+## Size of collision area
+@export var size = Vector3(3.372, 3.821, 1)
+## Time to teleport (in seconds, and this variable is optional)
+@export var time_teleport : int = 0
+
+## Get position of "Player" 
+@onready var position_player = get_node(Player)
+## Get position of "Target"
+@onready var position_target = get_node(Target)
+
+## Collision Shape
+var node_collision : CollisionShape3D
+## Timer node
+var node_time : Timer
```

Comment:
You can fully initiate this timer here, then it will never be undefined: `= Timer.new()`